src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,348 @@
+    },
+    model: {
+        contextId: "",
+        contextToCopy: "",
+        contextNames: [],
+        fieldErrors: ""
+    },
+    selectors: {
+        contextIdInput: "#pmt-add-context-name-input",
+        contextToCopySelect: "#pmt-context-to-copy-select"
+    },
+    bindings: {
+        contextIdInput: "contextId",
+        contextToCopySelect: "contextToCopy"
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog",
+            args: ["{that}.model", "{that}.applier", "{that}.closeDialog", "{arguments}.0"]
+        }
+    },
+    modelListeners: {
+        "fieldErrors": {
+            func: "{that}.reRender"
+        }
+    }
+});
+
+/**
+ * Dialog for adding a new preference set (context) to a preferences safe (prefset).
+ * Includes a dropdown that can be used to select a context to copy the
+ * initial settings from.
+ *
+ * @param {Object} model - Model object
+ * @param {changeApplier} applier - Change applier for model
+ * @param {Function} closeDialog - Zero-arg function to close dialog
+ * @param {event} event - Browser event triggered by the accept button.
+ */
+gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog = function (model, applier, closeDialog, event) {
+    // We don't want the form to actually submit the page, just to enable
+    // activating the submit button on Enter
+    event.preventDefault();
+
+    // Validate context name, eventually this will be replaced with
+    // our new json schema work.
+    if (model.contextId === "") {
+        applier.change("fieldErrors", "Please enter a name.");
+        return;
+    }
+    else if (!/^\w+$/.exec(model.contextId)) {
+        applier.change("fieldErrors", "Please use only alphanumeric characters.");
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        applier.change("fieldErrors", "");
+    }
+
+    // TODO validation to see if already exists, and determining
+    // the valid set of strings a preference set ID can take
+    var contextToCopy = model.contextToCopy;
+    var path = "flatPrefs.contexts." + model.contextId;
+    var newPrefSet = {};
+    // The select contains a blank option to start with a fresh set of values.
+    if (model.contextToCopy !== "" && model.flatPrefs.contexts[contextToCopy]) {
+        newPrefSet = fluid.copy(model.flatPrefs.contexts[contextToCopy].preferences);
+    }
+    applier.change(path, {
+        "name": model.contextId,
+        "preferences": newPrefSet
+    }, "ADD");
+    closeDialog();
+};
+
+/**
+ * Edit Preferences Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.editContextDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-editContext-dialog"
+    },
+    model: {
+        // Store so that we can fetch the context to copy to a new ID
+        originalContextId: "",
+        contextId: "",
```

Comment:
Remember to eliminate all use of the term "context" in favour of "prefs set name"

Reply:
CONTEXT has been COMPLETELY REMOVED!

Except for a few locations, where it is explicitly referencing the key 'contexts' as part of the preferences safe structure.  But otherwise... removed!